agent/eventhandler/task_handler.go

```diff
@@ -15,63 +15,96 @@ package eventhandler
 
 import (
 	"container/list"
+	"sync"
 	"time"
 
 	"github.com/aws/amazon-ecs-agent/agent/api"
 	"github.com/aws/amazon-ecs-agent/agent/utils"
+	"github.com/cihub/seelog"
 )
 
-var handler *taskHandler
+// Maximum number of tasks that may be handled at once by the TaskHandler
+const concurrentEventCalls = 3
```

Comment:
It's worth marking this down to revisit, especially since you're looking to continue modifying event submission.